python/pyarrow/tensor.pxi

```diff
@@ -207,6 +207,28 @@ shape: {0.shape}""".format(self)
                      &csparse_tensor))
         return pyarrow_wrap_sparse_coo_tensor(csparse_tensor)
 
+    @staticmethod
+    def from_pydata_sparse(obj, dim_names=None):
+        """
+        Convert pydata/sparse.COO to arrow::SparseCOOTensor
+        """
+        cdef shared_ptr[CSparseCOOTensor] csparse_tensor
+        cdef vector[int64_t] c_shape
+        cdef vector[c_string] c_dim_names
+
+        for x in obj.shape:
+            c_shape.push_back(x)
+        if dim_names is not None:
+            for x in dim_names:
+                c_dim_names.push_back(tobytes(x))
+
+        coords = np.require(obj.coords.T.view(), dtype='i8', requirements='F')
```

Comment:
I'm curious: why is the `view()` call necessary?

Reply:
I suppose it's not. Removing.

Reply:
We appear to be using it in other places (for scipy integration in `tensor.pxi`). Shall I remove those as well?

Reply:
If that doesn't fail the tests, then yes.

Reply:
Done. Seems to be ok.